packages/cascara/src/modules/DataCheckbox/DataCheckbox.js

```diff
@@ -30,8 +30,14 @@ const DataCheckbox = ({
   value,
   ...rest
 }) => {
-  const { isEditing, formMethods } = useContext(ModuleContext);
+  const {
+    formMethods,
+    record,
+    uniqueIdAttribute,
+    idOfRecordInEditMode,
+  } = useContext(ModuleContext);
```

Comment:
@lordpixel I think for this solution, we want the Data Modules to not have to be aware of all this information. Remember... they will also be used inside of a Form.

They should be able to look up the the nearest Module Provider and see if they should be editing or not. That can still be a boolean value.

I think all of this repeated code in all of these modules can actually get moved into the Row Context. We can make the row context react to changes in the parent context. That way all of the Data Modules can stay the same. All a module should care about is if it is editing or not. The logic for controlling that should exist outside of the module.

Remember that we have two nested contexts here. We can do some fancy work inside the nested context to translate that row to know if it should be editing or not.

Reply:
sure thing, let me work on it.